core/src/main/java/tech/pegasys/eth2signer/core/multikey/DirectoryLoader.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.eth2signer.core.multikey;
+
+import static java.util.Collections.emptySet;
+
+import tech.pegasys.eth2signer.core.multikey.metadata.parser.SignerParser;
+import tech.pegasys.eth2signer.core.signing.ArtifactSigner;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class DirectoryLoader {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final Path configsDirectory;
+  private final String fileExtension;
+  private final SignerParser signerParser;
+
+  public DirectoryLoader(
```

Comment:
No real benefit of a constructor since the operateOnFileSubset is only used from the static method

Reply:
That's true, but this means you don't need to pass everything _into_ everything, which made somethings nicer.